### ISSUE TITLE
Update po files for special ability notes' bullets

### ISSUE THREAD

**Issue by jostephd** [wesnoth/wesnoth#4297]
Split from https://github.com/wesnoth/wesnoth/pull/4231#issuecomment-524170557 @Wedge009

In 1.15 the special notes are shown as bullets (093fbc82fd7c2fd636a7f8fd4aeace55d025e64b). This invalidated translations. Can we somehow make translations valid again, for example, by inserting the newline + bullet into all translations?

I made the original change and I'd be happy to make some fix in all translations, but I'm not sure how to do that. Would I just mass-edit the po files? Can I just do this at any time? There was a pot-update recently (4830c76f45c5f5d6cc299dd850d468aee2c46b60) and we aren't in string freeze on master so I guess now would be a good time?

**Comment by jostephd** [comment]
@Wedge009 suggested to use pofix. However, we'd need to add a newline too; would that be a problem?

**Comment by jostephd** [comment]
Should I just commit the following change (but extended to all special notes and all po/pot files)?

```
diff --git a/po/wesnoth-httt/en_GB.po b/po/wesnoth-httt/en_GB.po
index c3612d4d940..771539b55a6 100644
--- a/po/wesnoth-httt/en_GB.po
+++ b/po/wesnoth-httt/en_GB.po
@@ -8823,15 +8823,12 @@ msgstr ""
 "defending."
 
 #: data/campaigns/Heir_To_The_Throne/utils/abilities.cfg:18
-#, fuzzy
-#| msgid ""
-#| " This unit's grasp of melee tactics allows adjacent allies to strike the "
-#| "first blow even when defending."
 msgid ""
 "\n"
 "• This unit's grasp of melee tactics allows adjacent allies to strike the "
 "first blow even when defending."
 msgstr ""
+"\n•"
 " This unit's grasp of melee tactics allows adjacent allies to strike the "
 "first blow even when defending."
 
```

**Comment by Wedge009** [comment]
For the mgstr, should use whatever was there before the po-update that made it fuzzy. But I believe that's essentially what needs to be done. I don't know if this could all be undone in a subsequent po-update, though, hence why I thought it should be done with pofix, but I'm no expert there either.

**Comment by jostephd** [comment]
That's a good point, I shouldn't remove the `fuzzy` marking if the string had been marked fuzzy prior to the pot-update. I'll await guidance.

**Comment by CelticMinstrel** [comment]
The reason for pofix is that if you just edit the po files directly, your work will be overwritten next time Ivanovic commits the latest translations.

So I wouldn't recommend anything that involves directly editing po files. If possible, I think you should find a way to use pofix. If it's not possible you could always try extending pofix so that it is possible. But, I think _adding_ a newline should be possible somehow...

**Comment by jostephd** [comment]
But pofix only changes msgid's, not msgstr's...

**Comment by CelticMinstrel** [comment]
Well, the fact remains that any attempt to manually update the po files risks being overwritten the next time Ivanovic pushes new translations.

**Comment by jostephd** [comment]
@ivanovic What would be the best way to handle https://github.com/wesnoth/wesnoth/commit/093fbc82fd7c2fd636a7f8fd4aeace55d025e64b in the po files?

**Comment by CelticMinstrel** [comment]
For the record, I'm not even entirely convinced that bullets are workable for all languages, so it might be best to just let the maintainers handle it themselves.

If we had the new `special_notes=` key that I'd been planning to add in various places for forever, this would be quite a bit easier &ndash; the bullets could be removed from the translatable strings and automatically added in by the help topic generator. (The bullet itself could then be made a separate translatable string if any translator complains it's not suitable to their language.)

**Comment by Wedge009** [comment]
I agree that would be a better solution: to have the bullets added by the special notes handler rather than tacked on to the string itself.

**Comment by CelticMinstrel** [comment]
Well, I think _part_ of my original plan was to add a way to place special notes directly in the unit definition (for the cases of special notes not directly linked to an ability, race, damage type, or whatever else which provides an obvious external location for it). So perhaps what we could do for now is add a new `[special_note]` tag to `[unit_type]` which could be used to append special notes, and use that for _all_ the special notes. (It has to be a tag even though there's only one subkey because we need to accept multiples of them and comma-separated notes would be pretty terrible to work with.)

**Comment by jostephd** [comment]
After #4305, there's still a difference translators will be faced with:

https://github.com/wesnoth/wesnoth/blob/4ce5b52c4e821e5d20b82f83e519e6402547df92/data/core/macros/special-notes.cfg#L10-L11

https://github.com/wesnoth/wesnoth/blob/ae73e5cd1546997b59e1156d8be3a7a9c0dbf79e/data/core/macros/special-notes.cfg#L17-L18

It's the leading space...

I guess we should remove the leading space from all translations simultaneously? I can look into that after string freeze starts tomorrow if somebody gives me the go ahead that it won't conflict with translators' work? Alternatively we can add `# po: A leading space was removed.` comments to the INTERNAL:* strings, so translators won't be confused by an apparently correct translation being marked fuzzy.

**Comment by Wedge009** [comment]
Doesn't the string freeze only apply to the text in cfg source and not the generated po files? Anyway, looks like we're back to the original problem of updating the po files in such a way as to not invalidate the existing translations, yet not risk having that fix be overwritten by the next pot-update.

**Comment by CelticMinstrel** [comment]
I don't think it matters that much if translations have an extra leading space… true, it would cause the strings to be marked fuzzy, but I'm pretty sure the translators would be able to quickly spot that the translation is still correct and remove the fuzzy label?

**Comment by jostephd** [comment]
@Wedge009 Yeah, I think I was confused about the order of events. A mass fix would need to happen before string freeze is announced to translators to prevent conflicts, that's what I was trying to say.

@CelticMinstrel I don't know... but if you're right, we can just close this issue.

**Comment by soliton-** [comment]
https://wiki.wesnoth.org/Translation_Maintainance_Commands#updating_po_files_for_committing pofix is run before committing translation updates.

**Comment by Wedge009** [comment]
celmin, if these were small strings it might be obvious that the change is trivial and fuzzies can be marked as done. But the notes are fairly long and translators may re-check the whole string if they're keen on being thorough, just in case there was any small change they might have missed.

I suppose we can just include an advice in the next translation e-mail that these strings have been trivially updated and don't need explicit reviewing.

**Comment by soliton-** [comment]
Pofix is used when you want to change the original string and not bother translators because the meaning did not change. That is the goal here AFAICT. Most important for this is to fix the `msgid` to the new string, i.e. remove the leading space.

Nice to have in this case would perhaps be to also automatically remove the space from the translation, i.e. `msgstr`. Since pofix normally replaces fixed strings and the translations are of course not fixed that would need additional logic which may be considered too fragile/complicated.

**Comment by jostephd** [comment]
The strings were last changed in 15f431785a55ae79e17f2b462c6e3bb8f18adc57, marking postponed until a pot-update is done.

**Comment by CelticMinstrel** [comment]
> That sounds like the best way to go.

This is the best way to go... _if_ that's actually the only change that was made to the strings. It likely is, but I don't know for sure.

**Comment by stevecotton** [comment]
Given the huge rework of special notes in #5811, I think this issue should be closed as outdated. The bullet-points are no longer part of the notes' text.

**Comment by Wedge009** [comment]
Agreed.

**Comment by CelticMinstrel** [comment]
Pretty sure that predates #5811 - didn't I remove the bullets in #4305?

**Comment by Wedge009** [comment]
IIRC the specific issue was the addition of bullet points into the translated text string itself was a problem, in that it unnecessarily flags strings to be re-translated even though the text itself was not changed. Since the bullet points are removed this shouldn't be an issue any more. As an example:

Current:
https://github.com/wesnoth/wesnoth/blob/480b6e3e7afb7a065bd0e9b0bf396b050fc34906/data/campaigns/Dead_Water/utils/specials.cfg#L8

Original:
https://github.com/wesnoth/wesnoth/blob/093fbc82fd7c2fd636a7f8fd4aeace55d025e64b/data/campaigns/Dead_Water/utils/specials.cfg#L5
